### PR TITLE
feat: `useMap`: allow resetting with provided value other then initial

### PR DESCRIPTION
**Pull request by mattleonowicz**
# Description

`useMap`: Add `setAll` to allow providing entirely new object

## Type of change

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [x] New feature _(non-breaking change which adds functionality)_
- [ ] **Breaking change** _(fix or feature that would cause existing functionality to not work as before)_

# Checklist
- [x] Read the [Contributing Guide](https://github.com/streamich/react-use/blob/master/CONTRIBUTING.md)
- [x] Perform a code self-review
- [x] Comment the code, particularly in hard-to-understand areas
- [x] Add documentation
- [x] Add hook's story at Storybook
- [x] Cover changes with tests
- [x] Ensure the test suite passes (`yarn test`)
- [x] Provide 100% tests coverage
- [x] Make sure code lints (`yarn lint`). Fix it with `yarn lint:fix` in case of failure.
- [x] Make sure types are fine (`yarn lint:types`).

### CODE REVIEW

**Comment by mattleonowicz** [comment]
Potentially if someone uses `reset` directly in an event handler like `onClick={reset}` then this would break his code, as the event would be passed as new object here. I'm not sure if that qualifies as a breaking change. Other solution would be to add a separate method like `resetWith`. Let me know what you think, I can change it accordingly.

File: src/useMap.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ const useMap = <T extends object = any>(initialMap: T = {} as T): [T, Actions<T>
           return rest as T;
         });
       },
-      reset: () => set(initialMap),
+      reset: (newMap = initialMap) => set(newMap),
```
Comment (streamich) [inline src/useMap.ts]: In this case, should you also keep track of the "new" `initialMap`?

```js
reset: (newMap = initialMapRef.current) => {
  initialMapRef.current = newMap;
  set(newMap);
},
```

**Comment by streamich** [comment]
Maybe instead of overloading `reset` function, we can add `put` function?

**Comment by mattleonowicz** [comment]
Yeah, maybe it would be better.
 In other languages it's common to see `put` and `putAll` for a Map,  so maybe we should use `setAll`, since we already have `set`.

**Pull request merged**